src/OrchardCore/OrchardCore.Media.Abstractions/Events/IMediaCreatingEventHandler.cs

```diff
@@ -0,0 +1,18 @@
+using System.IO;
+using System.Threading.Tasks;
+
+namespace OrchardCore.Media.Events
+{
+    /// <summary>
+    /// Event handler fired during creation of new media items.
+    /// </summary>
+    public interface IMediaCreatingEventHandler
+    {
+        /// <summary>
+        /// Allows a stream to be mutated during creation of media.
+        /// Any implementation must return a new stream,
+        /// which should be disposed by the caller.
+        /// </summary>
+        Task<Stream> MediaCreatingAsync(MediaCreatingContext context, Stream creatingStream);
```

Comment:
So if stream can be mutated then how about path? If original case was about converting files form one type to another then file extension in path also have to change in most cases.

Reply:
Yes, path is mutable. But the interface does not return that mutated path. I will think about this a bit more, now that I have the stream working how I want. Maybe it will have to move out of the store into the controller. Will see

Reply:
I change the `IFileStore` interface to return the (potentially mutated) path. 
Will think some more on this before deciding if it is the right thing